lib/datadog/tracing/contrib/sinatra/tracer_middleware.rb

```diff
@@ -29,6 +29,8 @@ def call(env)
               Tracing.continue_trace!(original_trace)
             end
 
+            return @app.call(env) if Sinatra::Env.datadog_span(env)
```

Comment:
Same as above: let's make sure short circuiting all Rack spans is safe. cc @lloeki